employees/tests/test_views.py

```diff
@@ -862,3 +862,104 @@ def test_view_should_display_author_reports_only_one_project(self):
         self.assertNotContains(response, report_from_other_project.description)
         self.assertEqual(response.status_code, 200)
         self.assertTemplateUsed(response, AuthorReportProjectView.template_name)
+
+
+class ReportSummaryTests(TestCase):
+    def setUp(self):
+        self.user = ManagerUserFactory()
+        project_1 = ProjectFactory()
+        project_1.members.add(self.user)
+        project_2 = ProjectFactory()
+        project_2.managers.add(self.user)
+        self.projects = [project_1, project_2]
+        self.current_time = timezone.now()
+        self.hours_per_report = 4
+        self.total_hours = 160
+        self._generate_user_reports_for_current_and_previous_month_with_uneven_total_hours()
+        self.expected_work_hours_stats = self._get_total_hours_per_project_and_percentage_from_month()
+
+    def test_report_list_should_accurately_evaluate_work_hours_statistics(self):
+        self.client.force_login(self.user)
+        url = reverse("custom-report-list", kwargs={"year": self.current_time.year, "month": self.current_time.month})
+
+        context_data = self.client.get(url).context_data
+        self._assert_projects_work_percentage_is_evaluated_accurately(context_data)
+
+    def test_author_report_list_should_accurately_evaluate_work_hours_statistics(self):
+        admin_user = AdminUserFactory()
+        self.client.force_login(admin_user)
+        url = reverse(
+            "author-report-list",
+            kwargs={"pk": self.user.pk, "year": self.current_time.year, "month": self.current_time.month},
+        )
+
+        context_data = self.client.get(url).context_data
+
+        self._assert_projects_work_percentage_is_evaluated_accurately(context_data)
+
+    def _assert_projects_work_percentage_is_evaluated_accurately(self, context_data):
+        self.assertTrue("projects_work_percentage" in context_data.keys())
+
+        projects_work_percentage = context_data["projects_work_percentage"]
+
+        self.assertEqual(self.expected_work_hours_stats, projects_work_percentage)
+
+    def _generate_user_reports_for_current_and_previous_month_with_uneven_total_hours(self):
+        reports_per_project, remaining_hours = self._strip_hours_between_projects()
```

Comment:
When you're returning tuple please make in openly. You can avoid any unexpected errors.